lightning/src/onion_message/messenger.rs

```diff
@@ -325,12 +325,18 @@ impl OnionMessageRecipient {
 
 /// The `Responder` struct creates an appropriate [`ResponseInstruction`]
 /// for responding to a message.
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Responder {
 	/// The path along which a response can be sent.
 	reply_path: BlindedPath,
 	path_id: Option<[u8; 32]>
```

Comment:
This is the `path_id` from the `BlindedPath` that we created for receiving the message. So, currently yes though we never set it. It's only in the `Responder` for logging purposes, which given the previous sentence means it's not very useful at the moment. 😛 

Though that is a good point considering we are likely going to change it to a `Vec` and start using it as part of #3085. I'm wondering how useful logging it really is unless we also log it when creating the blinded path, too.

Reply:
@TheBlueMatt Any thoughts on where we should go with this?

Reply:
It's useful for correlating an onion message to the blinded path it was sent over -- but only if the `path_id` of the latter is actually logged, too -- *and* with any responses to that onion message. But seeing that data encoded in the `path_id` will be given to the handler in #3085, then any logging responsibility could be moved to the handler instead, if desired, and use handler-specific concepts rather than opaque bytes.

I'm not sure where that leaves `Responder`. It would just a wrapper around `BlindedPath` with methods for converting to a `ResponseInstruction`. Those methods could be moved to `BlindedPath`, but that would add an `onion_message` module dependency to `blinded_path`, which doesn't feel right.  Maybe the wrapper is fine to keep?